tc/core/polyhedral/memory_promotion.cc

```diff
@@ -79,15 +119,21 @@ std::unique_ptr<TensorReferenceGroup> TensorReferenceGroup::makeSingleton(
 
 isl::set TensorReferenceGroup::approximateFootprint() const {
   auto scopedDomain = scopedAccesses().domain();
-  auto space = box.get_space();
+  auto space = approximation.box.get_space();
   auto accessed = isl::map::universe(space).intersect_domain(scopedDomain);
   auto lspace = isl::local_space(accessed.get_space().range());
 
-  for (size_t i = 0; i < dim(); ++i) {
-    auto dimLowerBound = lowerBound(i);
+  for (size_t i = 0; i < approximation.dim(); ++i) {
+    auto offset = approximation.lowerBound(i);
+    auto stride = approximation.stride(i);
+    auto strideOffset = approximation.strideOffset(i);
+    auto size = approximation.size(i);
     auto rhs = isl::aff(lspace, isl::dim_type::set, i);
-    isl::map partial = (isl::aff_map(dimLowerBound) <= rhs) &
-        (isl::aff_map(dimLowerBound + size(i)) > rhs);
+    auto lowerBound = offset * stride + strideOffset;
+    auto upperBound = (offset + size) * stride + strideOffset;
+    auto partial =
+        (isl::aff_map(lowerBound) <= rhs) & (isl::aff_map(upperBound) > rhs);
```

Comment:
Hmm.... if you need to spell it out like that (with a loop), then this
probably means there is some functionality missing in isl, but we
can clean it up later.